main.tf

```diff
@@ -2,18 +2,21 @@ locals {
   enabled                    = module.this.enabled ? 1 : 0
   parent_zone_record_enabled = var.parent_zone_record_enabled && module.this.enabled ? 1 : 0
   zone_name                  = local.parent_zone_record_enabled == 1 ? var.zone_name : replace(var.zone_name, ".$${parent_zone_name}", "")
+  public_zone                = try(length(var.private_hosted_zone_vpc_attachments), 0) == 0
 }
 
-data "aws_region" "default" {}
+data "aws_region" "default" {
+  count = local.enabled ? 1 : 0
+}
 
 data "aws_route53_zone" "parent_zone" {
-  count   = local.parent_zone_record_enabled
+  count   = local.parent_zone_record_enabled ? 1 : 0
   zone_id = var.parent_zone_id
   name    = var.parent_zone_name
 }
```

Comment:
Can we set private_zone = true here ?

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/route53_zone#private_zone